kalliope/core/OrderAnalyser.py

```diff
@@ -171,6 +187,31 @@ def _get_split_order_without_bracket(order):
         split_order = order.split()
         return split_order
 
+    @classmethod
+    def is_not_contain_matching(cls, user_order, signal_order, not_containing_words):
+        """
+        True if :
+            - all word in the user_order are present in the signal_order
+            - if not containing words in the order
+        :param user_order: order from the user
+        :param signal_order: order in the signal
+        :param not_containing_words: words which are not present in the order
+        :return: Boolean
+        """
+        logger.debug("[OrderAnalyser] is_not_contain_matching called with user_order: %s, signal_order: %s" % (user_order,
+                                                                                                          signal_order))
+        split_user_order = user_order.split()
```

Comment:
Following lines are similar that the normal match, this should be reused to avoid code duplication.